Index: b/lib/perfSONAR_PS/NPToolkit/Services/httpd.pm
===================================================================
--- a/lib/perfSONAR_PS/NPToolkit/Services/httpd.pm	2023-04-12 18:26:40.090293922 +0200
+++ b/lib/perfSONAR_PS/NPToolkit/Services/httpd.pm	2023-04-12 18:29:43.686724145 +0200
@@ -11,10 +11,10 @@
     my ( $self, %conf ) = @_;
 
     $conf{description}  = "Apache HTTP Server" unless $conf{description};
-    $conf{init_script} = "httpd" unless $conf{init_script};
-    $conf{process_names} = "httpd" unless $conf{process_names};
-    $conf{pid_files} = "/var/run/httpd/httpd.pid" unless $conf{pid_files};
-    $conf{package_names} = [ "httpd" ] unless $conf{package_names};
+    $conf{init_script} = "apache2" unless $conf{init_script};
+    $conf{process_names} = "apache2" unless $conf{process_names};
+    $conf{pid_files} = "/run/apache2/apache2.pid" unless $conf{pid_files};
+    $conf{package_names} = [ "apache2" ] unless $conf{package_names};
 
     $self->SUPER::init( %conf );
 
Index: b/lib/perfSONAR_PS/NPToolkit/Services/Cassandra.pm
===================================================================
--- a/lib/perfSONAR_PS/NPToolkit/Services/Cassandra.pm	2023-04-12 18:26:40.086858088 +0200
+++ b/lib/perfSONAR_PS/NPToolkit/Services/Cassandra.pm	2023-04-12 18:29:43.687927692 +0200
@@ -15,7 +15,7 @@
     #Cassandra does not properly create /var/run/cassandra directory on boot, only creates
     #  on install which then gets removed on reboot. Removing this for now.
     #$conf{pid_files} = "/var/run/cassandra/cassandra.pid" unless $conf{pid_files};
-    $conf{package_names} = [ "cassandra20" ] unless $conf{package_names};
+    $conf{package_names} = [ "cassandra" ] unless $conf{package_names};
 
     $self->SUPER::init( %conf );
 
Index: b/lib/perfSONAR_PS/NPToolkit/Services/Base.pm
===================================================================
--- a/lib/perfSONAR_PS/NPToolkit/Services/Base.pm	2023-04-12 18:29:43.390859659 +0200
+++ b/lib/perfSONAR_PS/NPToolkit/Services/Base.pm	2023-04-12 18:29:43.688745411 +0200
@@ -8,6 +8,7 @@
 use fields 'LOGGER', 'INIT_SCRIPT', 'PID_FILES', 'PROCESS_NAMES', 'DESCRIPTION', 'CAN_DISABLE', 'REGULAR_RESTART', 'PACKAGE_NAMES';
 use Dpkg::Control;
 use Dpkg::Version;
+use File::Basename;
 
 sub new {
     my ( $package ) = @_;
@@ -150,7 +151,7 @@
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    my $ret = system( "/usr/bin/systemctl", @param );
+    my $ret = system( "/bin/systemctl", @param );
 
     # restore stderr + stdout
     open(STDERR, ">&", $stderr);
@@ -169,7 +170,7 @@
 
     # Use systemd if available
     return $self->run_systemctl( 'is-enabled' => $self->{INIT_SCRIPT} )
-        if -x '/usr/bin/systemctl';
+        if -x '/bin/systemctl';
 
     # Check if the service is "on" in this run level.
     my $curr_runlevel;
@@ -184,17 +185,12 @@
 
     my $disabled = 1;
 
-    # turn off stderr
-    open(my $stderr, ">&STDERR");
-    open(STDERR, ">", File::Spec->devnull());
-
-    my $chkconfig_output = `/sbin/chkconfig --list $self->{INIT_SCRIPT}`;    
-
-    # restore stderr
-    open(STDERR, ">&", $stderr);
+    if (opendir(RC, "/etc/rc$curr_runlevel.d")) {
+        my @links = readdir(RC);
+        closedir(RC);
 
-    foreach my $line (split('\n', $chkconfig_output)) {
-        $disabled = 0 if ($line =~ /$curr_runlevel:on/);
+        my $service = basename($self->{INIT_SCRIPT});
+        $disabled = 0 if grep { /^S\d+$service$/ } @links;
     }
 
     return $disabled;
@@ -210,7 +206,7 @@
 
     # Use systemd if available
     return $self->run_systemctl( 'enable' => $self->{INIT_SCRIPT} )
-        if -x '/usr/bin/systemctl';
+        if -x '/bin/systemctl';
 
     # turn off stderr + stdout
     open(my $stderr, ">&STDERR");
@@ -218,9 +214,9 @@
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    system( "/sbin/chkconfig --del  " . $self->{INIT_SCRIPT} );
+    system("/usr/sbin/update-rc.d", basename($self->{INIT_SCRIPT}), "disable");
 
-    my $ret = system( "/sbin/chkconfig --add  " . $self->{INIT_SCRIPT} );
+    my $ret = system("/usr/sbin/update-rc.d", basename($self->{INIT_SCRIPT}), "enable");
 
     # restore stderr + stdout
     open(STDERR, ">&", $stderr);
@@ -239,7 +235,7 @@
 
     # Use systemd if available
     return $self->run_systemctl( 'disable' => $self->{INIT_SCRIPT} )
-        if -x '/usr/bin/systemctl';
+        if -x '/bin/systemctl';
 
     # turn off stderr + stdout
     open(my $stderr, ">&STDERR");
@@ -247,7 +243,7 @@
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    my $ret = system( "/sbin/chkconfig --del " . $self->{INIT_SCRIPT});
+    my $ret = system("/usr/sbin/update-rc.d", basename($self->{INIT_SCRIPT}), "disable");
 
     # restore stderr + stdout
     open(STDERR, ">&", $stderr);
@@ -266,7 +262,7 @@
 
     # Use systemd if available
     return $self->run_systemctl( $cmd => $self->{INIT_SCRIPT} )
-        if -x '/usr/bin/systemctl';
+        if -x '/bin/systemctl';
 
     # turn off stderr + stdout
     open(my $stderr, ">&STDERR");
@@ -274,7 +270,7 @@
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    my $shell_cmd = "/sbin/service " . $self->{INIT_SCRIPT} . " " . $cmd;
+    my $shell_cmd = "/usr/sbin/service " . $self->{INIT_SCRIPT} . " " . $cmd;
 
     $self->{LOGGER}->debug($shell_cmd);
 
Index: b/lib/perfSONAR_PS/Utils/Host.pm
===================================================================
--- a/lib/perfSONAR_PS/Utils/Host.pm	2023-04-12 18:26:40.114389256 +0200
+++ b/lib/perfSONAR_PS/Utils/Host.pm	2023-04-12 18:29:43.690208693 +0200
@@ -509,7 +509,7 @@
 sub get_ntp_info {
     my $ntp;
 
-    my $ntp_result = `/usr/sbin/ntpq -p`;
+    my $ntp_result = `/usr/bin/ntpq -p`;
 
     my @ntp_response = split /\n/, $ntp_result;
     
@@ -755,6 +755,7 @@
     my $result;
 
     my $is_el7 = 0;
+    my $is_debian = 0;
 
     if (    (   $os_info->{'distribution_name'}     =~ /^CentOS/
                 || $os_info->{'distribution_name'}  =~ /^Red Hat/
@@ -762,10 +763,19 @@
             )
             && $os_info->{'distribution_version'} =~ /^7\.\d/ ) {
                 $is_el7 = 1;
+    } elsif ($os_info->{'distribution_name'} =~ /^Debian/
+            || $os_info->{'distribution_name'}  =~ /^Ubuntu/ ) {
+            $is_debian = 1;
     }
     if ( $is_el7 ) {
         $result = `/bin/systemctl is-enabled yum-cron`;
 
+    } elsif ( $is_debian ) {
+        if (-e '/etc/apt/apt.conf.d/60unattended-upgrades-perfsonar') {
+            $result = "enabled";
+        } else {
+            $result = "disabled";
+        }
     } else {
         $result = `/etc/init.d/yum-cron status`;
     }
@@ -836,7 +846,7 @@
 sub _call_sysctl {
     my ($var_name) = @_;
 
-    my $result = `sysctl $var_name`;
+    my $result = `/sbin/sysctl $var_name`;
     if($?){
         return;
     }
Index: b/lib/perfSONAR_PS/NPToolkit/Services/YumCron.pm
===================================================================
--- a/lib/perfSONAR_PS/NPToolkit/Services/YumCron.pm	2023-04-12 18:26:40.090037120 +0200
+++ b/lib/perfSONAR_PS/NPToolkit/Services/YumCron.pm	2023-04-12 18:29:43.691159994 +0200
@@ -8,40 +8,42 @@
 sub init {
     my ( $self, %conf ) = @_;
 
-    $conf{description}  = "Yum Automatic Updates" unless $conf{description};
-    $conf{init_script} = "yum-cron" unless $conf{init_script};
-    $conf{process_names} = "yum" unless $conf{process_names};
-    $conf{pid_files} = "/var/lock/yum-cron.lock/pidfile" unless $conf{pid_files};
-    $conf{package_names} = [ "yum-cron" ] unless $conf{package_names};
+    $conf{description}  = "APT Automatic Updates" unless $conf{description};
+    $conf{package_names} = [ "unattended-upgrades" ] unless $conf{package_names};
 
     $self->SUPER::init( %conf );
 
     return 0;
 }
 
+sub disable_startup {
+    my ($self) = @_;
+
+    unlink '/etc/apt/apt.conf.d/60unattended-upgrades-perfsonar';
+}
+
 sub enable_startup {
     my ($self) = @_;
 
-    unless ($self->{INIT_SCRIPT}) {
-	$self->{LOGGER}->error("No init script specified for this service");
-	return -1;
+    my $action_file = '/etc/apt/apt.conf.d/60unattended-upgrades-perfsonar';
+    my $action_fd;
+    unless (open($action_fd, '>', $action_file)) {
+        $self->{LOGGER}->error("Could not opet $action_file: $!");
+        return -1;
     }
 
-    # turn off stderr + stdout
-    open(my $stderr, ">&STDERR");
-    open(my $stdout, ">&STDOUT");
-    open(STDERR, ">", File::Spec->devnull());
-    open(STDOUT, ">", File::Spec->devnull());
-
-    my $ret = system( "chkconfig --add  " . $self->{INIT_SCRIPT} );
-    #need to run chkconfig on
-    $ret = system( "chkconfig " . $self->{INIT_SCRIPT} . " on");
-    
-    # restore stderr + stdout
-    open(STDERR, ">&", $stderr);
-    open(STDOUT, ">&", $stdout);
-
-    return $ret;
+    print $action_fd <<"EOF";
+APT::Periodic::Update-Package-Lists "1";
+APT::Periodic::Download-Upgradeable-Packages "1";
+APT::Periodic::Unattended-Upgrade "1";
+APT::Periodic::AutocleanInterval "31";
+Unattended-Upgrade::Origins-Pattern:: "origin=perfSONAR";
+EOF
+
+    unless(close($action_fd)) {
+        $self->{LOGGER}->error("Could not close $action_file: $!");
+        return -1;
+    }
 }
 
 1;
Index: b/lib/perfSONAR_PS/NPToolkit/Services/NetworkBase.pm
===================================================================
--- a/lib/perfSONAR_PS/NPToolkit/Services/NetworkBase.pm	2023-04-12 18:26:40.088641562 +0200
+++ b/lib/perfSONAR_PS/NPToolkit/Services/NetworkBase.pm	2023-04-12 18:29:43.691851496 +0200
@@ -47,7 +47,7 @@
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    my $shell_cmd = "/sbin/service configure_nic_parameters start";
+    my $shell_cmd = "/usr/sbin/service configure_nic_parameters start";
 
     $self->{LOGGER}->debug($shell_cmd);
 
Index: b/lib/perfSONAR_PS/NPToolkit/Services/NTP.pm
===================================================================
--- a/lib/perfSONAR_PS/NPToolkit/Services/NTP.pm	2023-04-12 18:26:40.088308216 +0200
+++ b/lib/perfSONAR_PS/NPToolkit/Services/NTP.pm	2023-04-12 18:29:43.692515722 +0200
@@ -9,7 +9,7 @@
 sub init {
     my ( $self, %conf ) = @_;
 
-    $conf{init_script} = "ntpd" unless $conf{init_script};
+    $conf{init_script} = "ntp" unless $conf{init_script};
     $conf{description}  = "NTP" unless $conf{description};
     $conf{process_names} = "ntpd" unless $conf{process_names};
     $conf{pid_files} = "/var/run/ntpd.pid" unless $conf{pid_files};
