--- a/NPToolkit/Config/NTP.pm
+++ b/NPToolkit/Config/NTP.pm
@@ -35,8 +35,8 @@
 # These are the defaults for the current NPToolkit
 my %defaults = (
     ntp_conf          => "/etc/ntp.conf",
-    known_servers     => "/opt/perfsonar_ps/toolkit/etc/ntp_known_servers",
-    ntp_conf_template => "/opt/perfsonar_ps/toolkit/templates/config/ntp_conf.tmpl",
+    known_servers     => "/etc/perfsonar/toolkit/ntp_known_servers",
+    ntp_conf_template => "/usr/lib/perfsonar/templates/config/ntp_conf.tmpl",
     step_tickers_file => "/etc/ntp/step-tickers",
 );
 
