Index: perl-shared/lib/perfSONAR_PS/Utils/GeoLookup.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/Utils/GeoLookup.pm	2017-10-18 15:00:52.000000000 +0000
+++ perl-shared/lib/perfSONAR_PS/Utils/GeoLookup.pm	2017-11-01 18:25:27.000000000 +0000
@@ -71,8 +71,10 @@
 
             $record = $country_ipv6_db->country_code_by_addr_v6( $ip );
             $result->{'country'} = $record if ($record);
-            $record = $country_ipv6_db->country_name_by_addr_v6( $ip );
-            $result->{'country_full'} = $record if ($record);
+            if ( $country_ipv6_db->can("country_name_by_addr_v6") ) {
+                $record = $country_ipv6_db->country_name_by_addr_v6( $ip );
+                $result->{'country_full'} = $record if ($record);
+            }
         }
         else {
             warn "IP '".$ip."' was not detected as ipv4 or ipv6.";
