Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Config/NTP.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Config/NTP.pm	2017-12-18 17:09:47.000000000 +0000
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Config/NTP.pm	2017-12-19 14:52:18.000000000 +0000
@@ -37,7 +37,6 @@
     ntp_conf          => "/etc/ntp.conf",
     known_servers     => "/etc/perfsonar/toolkit/ntp_known_servers",
     ntp_conf_template => "/etc/perfsonar/toolkit/templates/ntp_conf.tmpl",
-    step_tickers_file => "/etc/ntp/step-tickers",
 );
 
 =head2 init({ ntp_conf_template => 0, known_servers => 0, ntp_conf => 0 })
@@ -111,10 +110,12 @@
         return (-1, "Problem saving list of known NTP servers");
     }
 
-    ($status, $res) = save_file( { file => $self->{STEP_TICKERS_FILE}, content => $ntp_step_tickers} );
-    if ( $status == -1 ) {
-        $self->{LOGGER}->error( "File save failed: " . $self->{STEP_TICKERS_FILE} );
-        return (-1, "Problem saving NTP configuration");
+    if ( $self->{STEP_TICKERS_FILE} ) {
+        ($status, $res) = save_file( { file => $self->{STEP_TICKERS_FILE}, content => $ntp_step_tickers} );
+        if ( $status == -1 ) {
+            $self->{LOGGER}->error( "File save failed: " . $self->{STEP_TICKERS_FILE} );
+            return (-1, "Problem saving NTP configuration");
+        }
     }
 
     if ( $parameters->{restart_services} ) {
