Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/MySQL.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/MySQL.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/MySQL.pm
@@ -9,8 +9,8 @@ sub init {
     my ( $self, %conf ) = @_;
 
     $conf{description}  = "MySQL Daemon" unless $conf{description};
-    $conf{init_script} = "mysqld" unless $conf{init_script};
-    $conf{process_names} = [ "mysql" ] unless $conf{process_names};
+    $conf{init_script} = "mysql" unless $conf{init_script};
+    $conf{process_names} = [ "mysqld" ] unless $conf{process_names};
     $conf{pid_files} = [ "/var/run/mysqld/mysqld.pid" ] unless $conf{pid_files};
     $conf{regular_restart} = 1;
     $conf{package_names} = [ "mysql-server" ] unless $conf{package_names};
Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/httpd.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/httpd.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/httpd.pm
@@ -11,10 +11,10 @@ sub init {
     my ( $self, %conf ) = @_;
 
     $conf{description}  = "Apache HTTP Server" unless $conf{description};
-    $conf{init_script} = "httpd" unless $conf{init_script};
-    $conf{process_names} = "httpd" unless $conf{process_names};
-    $conf{pid_files} = "/var/run/httpd/httpd.pid" unless $conf{pid_files};
-    $conf{package_names} = [ "httpd" ] unless $conf{package_names};
+    $conf{init_script} = "apache2" unless $conf{init_script};
+    $conf{process_names} = "apache2" unless $conf{process_names};
+    $conf{pid_files} = "/run/apache2/apache2.pid" unless $conf{pid_files};
+    $conf{package_names} = [ "apache2" ] unless $conf{package_names};
 
     $self->SUPER::init( %conf );
 
Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/Cassandra.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/Cassandra.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/Cassandra.pm
@@ -12,7 +12,7 @@ sub init {
     $conf{init_script} = "cassandra" unless $conf{init_script};
     $conf{process_names} = "java" unless $conf{process_names};
     $conf{pid_files} = "/var/run/cassandra/cassandra.pid" unless $conf{pid_files};
-    $conf{package_names} = [ "cassandra20" ] unless $conf{package_names};
+    $conf{package_names} = [ "cassandra" ] unless $conf{package_names};
 
     $self->SUPER::init( %conf );
 
Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/Base.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/Base.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/Base.pm
@@ -8,6 +8,7 @@ use File::Spec;
 use fields 'LOGGER', 'INIT_SCRIPT', 'PID_FILES', 'PROCESS_NAMES', 'DESCRIPTION', 'CAN_DISABLE', 'REGULAR_RESTART', 'PACKAGE_NAMES';
 use Dpkg::Control;
 use Dpkg::Version;
+use File::Basename;
 
 sub new {
     my ( $package ) = @_;
@@ -188,13 +189,21 @@ sub disabled {
     open(my $stderr, ">&STDERR");
     open(STDERR, ">", File::Spec->devnull());
 
-    my $chkconfig_output = `/sbin/chkconfig --list $self->{INIT_SCRIPT}`;    
+    my $insserv_output = `/sbin/insserv -s`;
 
     # restore stderr
     open(STDERR, ">&", $stderr);
 
-    foreach my $line (split('\n', $chkconfig_output)) {
-        $disabled = 0 if ($line =~ /$curr_runlevel:on/);
+    my $service = basename($self->{INIT_SCRIPT});
+    foreach my $line (split('\n', $insserv_output)) {
+        my @status = split(':', $line);
+
+        if ($status[0] eq 'S' &&
+            $status[3] eq $service &&
+            $status[2] =~ m/$curr_runlevel/)
+        {
+            $disabled = 0;
+        }
     }
 
     return $disabled;
@@ -218,9 +227,9 @@ sub enable_startup {
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    system( "/sbin/chkconfig --del  " . $self->{INIT_SCRIPT} );
+    system("/usr/sbin/update-rc.d", basename($self->{INIT_SCRIPT}), "disable");
 
-    my $ret = system( "/sbin/chkconfig --add  " . $self->{INIT_SCRIPT} );
+    my $ret = system("/usr/sbin/update-rc.d", basename($self->{INIT_SCRIPT}), "enable");
 
     # restore stderr + stdout
     open(STDERR, ">&", $stderr);
@@ -247,7 +256,7 @@ sub disable_startup {
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    my $ret = system( "/sbin/chkconfig --del " . $self->{INIT_SCRIPT});
+    my $ret = system("/usr/sbin/update-rc.d", basename($self->{INIT_SCRIPT}), "disable");
 
     # restore stderr + stdout
     open(STDERR, ">&", $stderr);
@@ -274,7 +283,7 @@ sub run_init {
     open(STDERR, ">", File::Spec->devnull());
     open(STDOUT, ">", File::Spec->devnull());
 
-    my $shell_cmd = "/sbin/service " . $self->{INIT_SCRIPT} . " " . $cmd;
+    my $shell_cmd = "/usr/sbin/service " . $self->{INIT_SCRIPT} . " " . $cmd;
 
     $self->{LOGGER}->debug($shell_cmd);
 
Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/NPAD.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/NPAD.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/NPAD.pm
@@ -24,9 +24,9 @@ sub init {
 
 sub is_installed{
     my ($self) = @_;
-    my $response = `rpm -qi npad`;
+    my $response = `dpkg -l npad 2>&1`;
 
-    my $substr = 'not installed';
+    my $substr = 'no packages found';
 
     if(index($response, $substr) == -1){
         #installed
Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/NDT.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/NDT.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/NDT.pm
@@ -53,9 +53,9 @@ sub kill {
 
 sub is_installed{
     my ($self) = @_;
-    my $response = `rpm -qi ndt`;
+    my $response = `dpkg -l ndt 2>&1`;
 
-    my $substr = 'not installed';
+    my $substr = 'no packages found';
 
     if(index($response, $substr) == -1){
         #installed
Index: perl-shared/lib/perfSONAR_PS/Utils/Host.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/Utils/Host.pm
+++ perl-shared/lib/perfSONAR_PS/Utils/Host.pm
@@ -496,7 +496,7 @@ sub discover_primary_address {
 sub get_ntp_info {
     my $ntp;
 
-    my $ntp_result = `/usr/sbin/ntpdc -p`;
+    my $ntp_result = `/usr/bin/ntpdc -p`;
 
     my @ntp_response = split /\n/, $ntp_result;
     
@@ -700,16 +700,11 @@ sub get_tcp_configuration {
 }
 
 sub is_auto_updates_on{
-
     my ($self) = @_;
 
-    my $enabled = "enabled";
-
-    my $result = `/etc/init.d/yum-cron status`;
-
-    if(index($result, $enabled) != -1){
+    if(-e '/etc/cron-apt/action.d/5-upgrade'){
         return 1;
-    }else{
+    } else {
         return 0;
     }
 }
@@ -773,7 +768,7 @@ sub is_web_url {
 sub _call_sysctl {
     my ($var_name) = @_;
 
-    my $result = `sysctl $var_name`;
+    my $result = `/sbin/sysctl $var_name`;
     if($?){
         return;
     }
Index: perl-shared/lib/perfSONAR_PS/NPToolkit/Services/YumCron.pm
===================================================================
--- perl-shared.orig/lib/perfSONAR_PS/NPToolkit/Services/YumCron.pm
+++ perl-shared/lib/perfSONAR_PS/NPToolkit/Services/YumCron.pm
@@ -8,40 +8,36 @@ use base 'perfSONAR_PS::NPToolkit::Servi
 sub init {
     my ( $self, %conf ) = @_;
 
-    $conf{description}  = "Yum Automatic Updates" unless $conf{description};
-    $conf{init_script} = "yum-cron" unless $conf{init_script};
-    $conf{process_names} = "yum" unless $conf{process_names};
-    $conf{pid_files} = "/var/lock/yum-cron.lock/pidfile" unless $conf{pid_files};
-    $conf{package_names} = [ "yum-cron" ] unless $conf{package_names};
+    $conf{description}  = "APT Automatic Updates" unless $conf{description};
+    $conf{package_names} = [ "cron-apt" ] unless $conf{package_names};
 
     $self->SUPER::init( %conf );
 
     return 0;
 }
 
+sub disable_startup {
+    my ($self) = @_;
+
+    unlink '/etc/cron-apt/action.d/5-upgrade';
+}
+
 sub enable_startup {
     my ($self) = @_;
 
-    unless ($self->{INIT_SCRIPT}) {
-	$self->{LOGGER}->error("No init script specified for this service");
-	return -1;
+    my $action_file = '/etc/cron-apt/action.d/5-upgrade';
+    my $action_fd;
+    unless (open($action_fd, '>', $action_file)) {
+        $self->{LOGGER}->error("Could not opet $action_file: $!");
+        return -1;
     }
 
-    # turn off stderr + stdout
-    open(my $stderr, ">&STDERR");
-    open(my $stdout, ">&STDOUT");
-    open(STDERR, ">", File::Spec->devnull());
-    open(STDOUT, ">", File::Spec->devnull());
-
-    my $ret = system( "chkconfig --add  " . $self->{INIT_SCRIPT} );
-    #need to run chkconfig on
-    $ret = system( "chkconfig " . $self->{INIT_SCRIPT} . " on");
-    
-    # restore stderr + stdout
-    open(STDERR, ">&", $stderr);
-    open(STDOUT, ">&", $stdout);
+    print $action_fd "-q -q dist-upgrade\n";
 
-    return $ret;
+    unless(close($action_fd)) {
+        $self->{LOGGER}->error("Could not close $action_file: $!");
+        return -1;
+    }
 }
 
 1;
